From: Maciej Fijalkowski <maciej.fijalkowski@intel.com>
Date: Tue, 1 Oct 2024 22:06:05 +0200
Subject: bpf: Remove unused macro
Patch-mainline: v6.13-rc1
Git-commit: 8f5b408d7661e33157b16c4e4d232f483e8e4f79
References: jsc#PED-14653

Commit 7aebfa1b3885 ("bpf: Support narrow loads from bpf_sock_addr.user_port")
removed one and only SOCK_ADDR_LOAD_OR_STORE_NESTED_FIELD callsite but kept
the macro. Remove it to clean up the code base. Found while getting lost in
the BPF code.

Signed-off-by: Maciej Fijalkowski <maciej.fijalkowski@intel.com>
Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
Link: https://lore.kernel.org/bpf/20241001200605.249526-1-maciej.fijalkowski@intel.com
Acked-by: Hoyeon Lee <hoyeon.lee@suse.com>
---
 net/core/filter.c |    4 ----
 1 file changed, 4 deletions(-)

--- a/net/core/filter.c
+++ b/net/core/filter.c
@@ -10227,10 +10227,6 @@ static u32 xdp_convert_ctx_access(enum b
 		}							       \
 	} while (0)
 
-#define SOCK_ADDR_LOAD_OR_STORE_NESTED_FIELD(S, NS, F, NF, TF)		       \
-	SOCK_ADDR_LOAD_OR_STORE_NESTED_FIELD_SIZE_OFF(			       \
-		S, NS, F, NF, BPF_FIELD_SIZEOF(NS, NF), 0, TF)
-
 static u32 sock_addr_convert_ctx_access(enum bpf_access_type type,
 					const struct bpf_insn *si,
 					struct bpf_insn *insn_buf,
